src/store/modules/rate/rate-actions.ts

```diff
@@ -0,0 +1,68 @@
+import { ActionTree } from 'vuex'
+import { RootState } from '@/store/types'
+import rateInfoClient from '@/lib/nodes/rate-info-service'
+import { RateState } from '@/store/modules/rate/types.ts'
+
+export let interval: NodeJS.Timeout
+
+const UPDATE_RATES_INTERVAL = 90000
+
+export const actions: ActionTree<RateState, RootState> = {
+  getAllRates({ commit }) {
+    return new Promise((resolve, reject) => {
+      rateInfoClient
+        .getAllRates()
+        .then(({ result, success }) => {
+          if (success) {
+            commit('setRates', result)
+          }
+          commit('loadRates')
+          resolve(result)
+        })
+        .catch((err) => {
+          reject(err)
+        })
+    })
+  },
```

Comment:
There no need in wrapping in `new Promise()`.

1. Prefix function with `async`

2. Use as:

```js
const { result, success } = await rateInfoClient.getAllRates()
if (success) {
  commit('setRates', result)
}
```

Reply:
Similar for `getHistoryRates`